### DISCUSSION TITLE
Is there a way to restrict variables values ?

### DISCUSSION THREAD

**Discussion by SimTheFool** [go-task/task#913] [Q&A]
Hello !

I would like to add some control on variables passed on my tasks. For example, let's I've a FILE variable, I would like it to be specified as "client.js" or "client-standalone.js" but "server.js" wouldn't be allowed and would exit my running task with ideally an error message.

For more visual example, that's what I'm after:

```
vars:
  FILE: '{{default "client.js" .T}}'
     in: [clientjs, client-standalone.js]
     error: Error, your file cannot be processed
```

Any idea about what's the way to go here ?

**Comment by dh7892** [comment]
I'd love this too. 

I came across this post when checking to see if anyone had asked the question I was going to ask, which was:

Is there any way to specify that a task (or sub-task) must have certain vars defined or it should abort? (I think not as I can't see anything in the docs).

I don't mind the pattern of passing info around in vars, rather than arguments but without having a way to easily check that those vars have been defined can lead to a lot of brittle behaviour.

What SimTheFool suggests here would be even better and would cover my needs as well as adding even more
